packages/shared/src/lib/core/market/apis/coingecko.api.ts

```diff
@@ -0,0 +1,53 @@
+import { DEFAULT_APPLICATION_JSON_REQUEST_OPTIONS } from '@core/utils'
+import { MARKET_API_BASE_URL } from '../constants'
+import { CoinGeckoApiEndpoint, MarketCoinId, MarketCurrency } from '../enums'
+import { MarketCoinPrices } from '../types'
+import { CoinGeckoCoin, CoinGeckoCoinShimmerEVM } from '../interfaces'
+
+export class CoinGeckoApi {
+    static async makeRequest<T>(endpoint: string, queryParams?: string): Promise<T> {
+        try {
+            const response = await fetch(
+                `${MARKET_API_BASE_URL}${endpoint}?${queryParams ?? ''}`,
+                DEFAULT_APPLICATION_JSON_REQUEST_OPTIONS
+            )
+            const data = await response.json()
+            return <T>data
+        } catch (err) {
+            console.error(err)
+            throw err
+        }
+    }
+
+    static async getSimplePrices(ids: MarketCoinId[], vsCurrencies: MarketCurrency[]): Promise<MarketCoinPrices> {
+        const queryParams = buildQueryParametersFromObject({
+            ids: ids.join(','),
+            vs_currencies: vsCurrencies.join(','),
+        })
+        return this.makeRequest<MarketCoinPrices>(CoinGeckoApiEndpoint.SIMPLE_PRICE, queryParams)
+    }
+
+    static async getCoinsList(include_platform: boolean): Promise<CoinGeckoCoin[]> {
+        const queryParams = buildQueryParametersFromObject({
+            include_platform,
+        })
+        return this.makeRequest<CoinGeckoCoin[]>(CoinGeckoApiEndpoint.COINS_LIST, queryParams)
+    }
+
+    static async getFilteredCoinsList(): Promise<CoinGeckoCoinShimmerEVM[]> {
+        const coinsList = await this.getCoinsList(true)
+        return coinsList.filter((coin) =>
+            Object.keys(coin.platforms).includes('shimmer_evm')
```

Comment:
Pass this in as a filter and then store the values for coingecko network ids in an enum